forem/forem
# Theme data update script, remove theme choices from UI

_NOTE: According to internal discussions we'll probably post a notice about this to post forem.dev and dev.to which will stay online for about a week before pulling the trigger, which means this PR will be merged when I'm on leave. I'll set it to draft until then and add the "merge by any core" label._

## What type of PR is this? (check all applicable)


## Description

This PR will migrate all existing users to either the light or dark theme according to the following rules described by @ludwiczakpawel in #14835:

| Old theme | New theme | `prefer_os_color_scheme` |
| --- | --- | --- |
| Default | Light | true |
| Minimal | Light | false |
| Night | Dark | false |
| Pink | Light | true |
| Ten X Hacker | Dark | false |

Notes: 

* The `prefer_os_color_scheme` setting is not effective yet, this is purely a data migration.
* Since data update scripts run ~10 minutes after a deploy all old themes are still available so users who have one of them selected won't experience any UI glitches. However, they can only switch between light and dark going forward and there's no check box for OS sync yet.

## Related Tickets & Documents

#14835

## QA Instructions, Screenshots, Recordings

![image](https://user-images.githubusercontent.com/47985/139377523-0f72685b-1549-47b5-8dd3-403f2bf6fde7.png)

* Enable a different theme for your user
* Run the DUS
* Make sure the expected changes happened
* Make sure only the light and dark themes are available in the theme selector (http://localhost:3000/settings/customization)

### UI accessibility concerns?

None

## Added/updated tests?


## [Forem core team only] How will this change be communicated?
